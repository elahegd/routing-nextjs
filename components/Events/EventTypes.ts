export interface EventItemInterface {
  id: string;
  title: string;
  description?: string;
  location: string;
  date: string;
  image: string;
  isFeatured?: boolean;
}
export type EventListType = EventItemInterface[];
