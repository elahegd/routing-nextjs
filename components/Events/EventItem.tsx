import React, { FC } from "react";

import { EventItemInterface } from "./EventTypes";
import classes from "./EventItem.module.css";
import Button from "../Ui/Button";
import DateIcon from "../Icons/date-icon";
import AddressIcon from "../Icons/address-icon";
import ArrowRightIcon from "../Icons/arrow-right-icon";

const EventItem: FC<EventItemInterface> = ({
  id,
  title,
  location,
  date,
  image,
}) => {
  const humanReadableDate = new Date(date).toLocaleDateString("en-US", {
    year: "numeric",
    month: "long",
    day: "numeric",
  });

  const formattedAddress = location.replace(", ", "\n");

  const exploreLink = `/events/${id}`;

  return (
    <li className={classes.item}>
      <img src={`/${image}`} alt={title} />
      <div className={classes.content}>
        <div className={classes.summary}>
          <h2>{title}</h2>
          <div className={classes.date}>
            <DateIcon />
            <time>{humanReadableDate}</time>
          </div>
          <div className={classes.address}>
            <AddressIcon />
            <address>{formattedAddress}</address>
          </div>
        </div>
        <div className={classes.actions}>
          <Button link={exploreLink}>
            <span>Explore Event</span>
            <span className={classes.icon}>
              <ArrowRightIcon />
            </span>
          </Button>
        </div>
      </div>
    </li>
  );
};

export default EventItem;
