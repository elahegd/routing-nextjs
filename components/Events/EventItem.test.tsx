import React from "react";
import { render } from "@testing-library/react";

import EventItem from "./EventItem";
import { EventItemInterface } from "./EventTypes";

describe("Render and check props", () => {
  const mockEventitem: EventItemInterface = {
    id: "e3",
    title: "Networking for extroverts",
    location: "My Street 12, 10115 Broke City",
    date: "2022-04-10",
    image: "images/extrovert-event.jpg",
  };

  it("Render component with exact props", () => {
    const { id, title, location, date, image } = mockEventitem;

    const { asFragment } = render(
      <EventItem
        id={id}
        title={title}
        location={location}
        date={date}
        image={image}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
