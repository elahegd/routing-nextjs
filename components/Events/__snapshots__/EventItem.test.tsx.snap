// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render and check props Render component with exact props 1`] = `
<DocumentFragment>
  <li
    class="item"
  >
    <img
      alt="Networking for extroverts"
      src="/images/extrovert-event.jpg"
    />
    <div
      class="content"
    >
      <div
        class="summary"
      >
        <h2>
          Networking for extroverts
        </h2>
        <div
          class="date"
        >
          <svg
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 7V3m8 4V3m-9 8h10M5 21h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v12a2 2 0 002 2z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <time>
            April 10, 2022
          </time>
        </div>
        <div
          class="address"
        >
          <svg
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.657 16.657L13.414 20.9a1.998 1.998 0 01-2.827 0l-4.244-4.243a8 8 0 1111.314 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M15 11a3 3 0 11-6 0 3 3 0 016 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <address>
            My Street 12
10115 Broke City
          </address>
        </div>
      </div>
      <div
        class="actions"
      >
        <a
          class="btn"
          href="/events/e3"
        >
          <span>
            Explore Event
          </span>
          <span
            class="icon"
          >
            <svg
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M12.293 5.293a1 1 0 011.414 0l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-2.293-2.293a1 1 0 010-1.414z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </a>
      </div>
    </div>
  </li>
</DocumentFragment>
`;
